internal/provider/data_source_http.go

```diff
@@ -49,6 +54,30 @@ your control should be treated as untrustworthy.`,
 				Optional: true,
 			},
 
+			"request_timeout": {
+				Description: "The request timeout in milliseconds.",
+				Type:        types.Int64Type,
+				Optional:    true,
+				Validators: []tfsdk.AttributeValidator{
+					int64validator.AtLeast(1),
+				},
+			},
+
+			"retry": {
+				Description: "Retry request configuration.",
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
```

Comment:
Since we're still on protocol version 5, this will need to be migrated to `Blocks` for now -- `Nesting: tfsdk.BlockNestingModeList` with `MaxItems: 1` would be the closest we can get there. 🙁 (It could technically be an Object attribute, but then it'd lose all the "nice" things the framework can provide for the nested attributes).